trezor-common/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Common code used by Trezor firware with no-std
+
+#![no_std]
+
+use num_derive::FromPrimitive;
+use parity_scale_codec::{Decode, Encode};
+use strum::{EnumDiscriminants, EnumIter};
+
+/// Specifies which parts of the transaction a signature commits to.
+///
+/// The values of the flags are the same as in Bitcoin.
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Ord, PartialOrd, Encode, Decode)]
+pub struct SigHashType(u8);
+
+impl SigHashType {
+    pub const ALL: u8 = 0x01;
+    pub const NONE: u8 = 0x02;
+    pub const SINGLE: u8 = 0x03;
+    pub const ANYONECANPAY: u8 = 0x80;
+
+    pub const MASK_OUT: u8 = 0x7f;
+    pub const MASK_IN: u8 = 0x80;
+
+    pub fn get(&self) -> u8 {
+        self.0
+    }
+}
+
+type UnsignedIntType = u128;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub struct Amount {
+    #[codec(compact)]
+    atoms: UnsignedIntType,
+}
+
+impl Amount {
+    pub const MAX: Self = Self::from_atoms(UnsignedIntType::MAX);
+    pub const ZERO: Self = Self::from_atoms(0);
+
+    pub const fn from_atoms(v: UnsignedIntType) -> Self {
+        Amount { atoms: v }
+    }
+
+    pub const fn into_atoms(&self) -> UnsignedIntType {
+        self.atoms
+    }
+
+    pub fn from_bytes_be(bytes: &[u8]) -> Option<Self> {
+        bytes
+            .try_into()
+            .ok()
+            .map(|b| Self::from_atoms(UnsignedIntType::from_be_bytes(b)))
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum OutputValue {
+    Coin(Amount),
+    TokenV0,
+    TokenV1(H256, Amount),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, EnumDiscriminants)]
+#[strum_discriminants(name(OutputTimeLockIndex), derive(EnumIter, FromPrimitive))]
```

Comment:
As I can see, `OutputTimeLockIndex` has nothing to do with `OutputTimeLock`, same as it was for `AccountCommandIndex` and `OutPointSourceIdIndex`. But it must be conmsistent with `lock_type` returned by `get_lock` in the trezor repo. So it makes sense to do the same for it as well - move it to rust code in the trezo repo, add same enum to the .proto file, write tests that ensures their consistency.

P.S. probably these extra protobuf enums should go to a separate file, because they have nothing to do with the protocol and only serve as a way to have the same constants in python and rust.